### PR TITLE
Fix secure sockets after FD table introduction

### PR DESCRIPTION
**Pull request by rlubos**
This PR fixes secure sockets implementation by removing invalid typecasts between socket descriptor and `net_context`. Note, that this PR does not add full support for POSIX APIs for secure sockets (like read/write), it's just a bugfix.

With FD table introduction, net_context can no longer be reached by typecasting socket descriptor to `net_context`, and vice versa. Instead, file descriptor API have to be used to reach the `net_context`.

As reverse conversion is not possible (`net_context` -> file descriptor), we need to provide file descriptor to functions that call `zsock_*` API (i.e. mbedTLS bio functions).

Additionally, second commit fixes a bug with `getsockopt`/`setsockopt` reuturn values, spotted during rework.

### CODE REVIEW

File: subsys/net/lib/sockets/sockets_tls.c [hunk 1]
```diff
@@ -143,6 +144,11 @@ static struct k_mutex context_lock;
 #define IS_LISTENING(context) (net_context_get_state(context) == \
 			       NET_CONTEXT_LISTENING)
 
+static inline struct net_context *sock_to_net_ctx(int sock)
+{
+	return z_get_fd_obj(sock, NULL, ENOTSOCK);
```
Comment (pfalcon) [inline subsys/net/lib/sockets/sockets_tls.c]: I hope we both understand that passing vtable=NULL is a hack.

File: subsys/net/lib/sockets/sockets_tls.c [hunk 2]
```diff
@@ -1119,7 +1142,7 @@ int ztls_socket(int family, int type, int proto)
 
 	if (tls_proto != 0) {
 		/* If TLS protocol is used, allocate TLS context */
-		struct net_context *context = INT_TO_POINTER(sock);
+		struct net_context *context = sock_to_net_ctx(sock);
```
Comment (pfalcon) [inline subsys/net/lib/sockets/sockets_tls.c]: No NULL check here, unlike done in other places.

**Review by pfalcon — APPROVED** [review]
Let me invoke "Trivial patch" spell and approve.

**Comment by nashif** [comment]
> Let me invoke "Trivial patch" spell and approve.

note that trivial sets minimum review time, it does not mean it needs to be expedited. hope that was clear in the proposal.

File: subsys/net/lib/sockets/sockets_tls.c (hunk 1 above)
Comment (rlubos) [reply subsys/net/lib/sockets/sockets_tls.c]: Yes, we do. I'm already working on a solution for secure sockets, according to [this](https://github.com/zephyrproject-rtos/zephyr/pull/10429#discussion_r228254959) comment.

**Comment by pfalcon** [comment]
@nashif :

> note that trivial sets minimum review time, it does not mean it needs to be expedited. hope that was clear in the proposal.

For me, those are quite related. Having "Trivial" category from my PoV is effectively a pledge along the lines of: "Dear fellow developer, you see that we're stuck at 250+ unprocessed patches, and that number is only growing. Then you may wonder, if you see any c%ap in our code, does it make sense to try to fix it, or it's useless, and it's better to just stick to work on what your boss told you. We have good news for you - if you submit us an easy to review patch, we'll actually try review it as soon as 4 hrs."

If that's (improving the process for contributors, again, as was announced by you 3 weeks ago) not the intention behind all this, then well, d'oh.

**Pull request merged**